### PR TITLE
Support implicit VB.NET declaration value in SE

### LINKED ISSUE #5469: Support implicit VB.NET declaration value in SE
```
Dim B As Boolean
```
is equivalent to
```
var b = false;
```
and the implicit assignment operation is not present from Roslyn. We need to create the constraint based on `LocalLifeTimeRegion`

While I've asked Roslyn to provide this feature, it will not be backported to existing versions of Roslyn: 
https://github.com/dotnet/roslyn/issues/60056

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #5469

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs [hunk 1]
```diff
@@ -19,23 +19,31 @@
...
 {
     internal class ConstantCheck : SymbolicCheck
     {
         protected override ProgramState PreProcessSimple(SymbolicContext context) =>
-            GetConstraint(context.Operation.Instance) is { } constraint
+            ConstraintFromValue(context.Operation.Instance.ConstantValue.Value) is { } constraint
                 ? context.SetOperationConstraint(constraint)
                 : context.State;
 
-        private static SymbolicConstraint GetConstraint(IOperation operation) =>
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs:33]: I suggest moving all these helper methods to a separate file with other helpers. They are not tightly coupled with the constant check.

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs:33]: I'd keep them here, as their behavior is driven by what `ConstantCheck` is doing. And the other usage should only follow what the ConstantCheck it does.

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs:33]: Only one of them should be public.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**